src/Sqids/SqidsEncoder.cs

```diff
@@ -224,8 +224,10 @@ private string Encode(ReadOnlySpan<int> numbers, int increment = 0)
 		offset = (numbers.Length + offset) % _alphabet.Length;
 		offset = (offset + increment) % _alphabet.Length;
 
+		char[]? pooledArray = null;
+
 		Span<char> alphabetTemp = _alphabet.Length * sizeof(char) > MaxStackallocSize // NOTE: We multiply the number of characters by the size of a `char` to get the actual amount of memory that would be allocated.
-			? new char[_alphabet.Length]
+			? (pooledArray = ArrayPool<char>.Shared.Rent(_alphabet.Length)).AsSpan(_alphabet.Length)
```

Comment:
The AsSpan is to ensure the correct length. As Rent most likely will give us an Array that is too big and avoids the implicit conversion.